### DISCUSSION TITLE
Execute queries without registering Data Frames / Arrow Tables in R?

### DISCUSSION THREAD

**Discussion by eitsupi** [duckdb/duckdb#6771] [Q&A] [answered]
It is really useful to be able to query `polars.DataFrame` etc. without explicit registration in the Python client.

```python
import duckdb
import polars as pl
df = pl.DataFrame({'a': 42})
duckdb.sql('select * from df').pl()
```

Are there plans to implement similar functionality in the R client?
Or is this difficult due to performance issues, etc.?

For now, it seems that connections need to be explicitly created and registered with the DB using the `duckdb_register()` function.

**Comment by hannes** [comment, marked as answer]
Yes, that auto-registration is not that hard to add in R, but we have not gotten around to it.

**Reply by eitsupi** [reply]
Thanks for your answer!
I look forward to seeing that implemented.
